server/events/comment_parser.go

```diff
@@ -65,9 +65,10 @@ type CommentBuilder interface {
 
 // CommentParser implements CommentParsing
 type CommentParser struct {
```

Comment:
I don't think this logic should be added here. The CommentParser's job is to parse comments. Not to determine what to do with them.

Instead, I think the best place for this logic is here: https://github.com/runatlantis/atlantis/blob/master/server/events/command_runner.go#L143

Reply:
The code should look like:
```go
	if c.DisableApplyAll && cmd.Name == models.ApplyCommand && !cmd.IsForSpecificProject() {
		if err := c.VCSClient.CreateComment(baseRepo, pullNum, applyAllDisabledComment); err != nil {
			log.Err("unable to comment on pull request: %s", err)
		}
		return
	}
```

Reply:
You will also need to add a test case